qunicorn_core/core/pilotmanager/pilot_manager.py

```diff
@@ -48,6 +52,37 @@ def update_devices_from_provider(provider_id: int, token: Optional[str]):
     pilot: Pilot = get_matching_pilot(provider.name)
     pilot.save_devices_from_provider(token)
 
+    updated_provider = ProviderDataclass.get_by_id_or_404(provider_id)
+    qprov_provider_name = provider_name_map[updated_provider.name]
+    qprov_root_url = current_app.config.get("QPROV_URL")
+
+    if qprov_root_url is None:
+        current_app.logger.info("QPROV_URL not set, skipping QPROV update")
+        return
+
+    response = requests.get(f"{qprov_root_url}/providers")
+    response.raise_for_status()
+    qprov_providers = response.json()["_embedded"]["providerDtoes"]
+
+    for qprov_provider in qprov_providers:
+        if qprov_provider["name"] == qprov_provider_name:
+            qprov_id = qprov_provider["id"]
+            provider.qprov_id = uuid.UUID(qprov_id)
+
+            response = requests.get(f"{qprov_root_url}/qprov/providers/{qprov_id}/qpus")
```

Comment:
this url is resulting in .../qprov/qprov/...
`response = requests.get(f"{qprov_root_url}/providers/{qprov_id}/qpus")`
works for me